controllers/awsmachine_controller.go

```diff
@@ -263,16 +275,20 @@ func (r *AWSMachineReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Ma
 	)
 }
 
-func (r *AWSMachineReconciler) reconcileDelete(machineScope *scope.MachineScope, clusterScope cloud.ClusterScoper, ec2Scope scope.EC2Scope, elbScope scope.ELBScope) (ctrl.Result, error) {
+func (r *AWSMachineReconciler) reconcileDelete(machineScope *scope.MachineScope, clusterScope cloud.ClusterScoper, ec2Scope scope.EC2Scope, elbScope scope.ELBScope, objectStoreScope scope.S3Scope) (ctrl.Result, error) {
 	machineScope.Info("Handling deleted AWSMachine")
 
 	ec2Service := r.getEC2Service(ec2Scope)
 
-	if machineScope.UseSecretsManager() {
-		if err := r.deleteEncryptedBootstrapDataSecret(machineScope, clusterScope); err != nil {
-			machineScope.Error(err, "unable to delete machine")
-			return ctrl.Result{}, err
-		}
+	if err := r.deleteEncryptedBootstrapDataSecret(machineScope, clusterScope); err != nil {
+		machineScope.Error(err, "unable to delete machine")
+		return ctrl.Result{}, err
+	}
+
+	objectStoreSvc := r.getObjectStoreService(objectStoreScope)
```

Comment:
Do we need to make this conditional as `objectStoreScope` is nil for managed control plane clusters?

Reply:
I think you're right, but I don't see any tests failing right now, so it doesn't seem strictly necessary?

Reply:
I've just kicked off the EKS e2e tests which i expect will fail because of this. I thought i'd kicked them off earlier but started the wrong test...doh.

Reply:
I've made it optional and now both e2e tests suites are passing. I didn't add tests though, any suggestions for those?